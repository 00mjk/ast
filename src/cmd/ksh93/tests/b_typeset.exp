# vim: set filetype=expect:
#
# Tests of `typeset` keyword/builtin.
set pid [spawn $ksh]
expect_prompt
# Terminal rows and columns may default to zero so ensure sane values.
send "stty rows 24 cols 80\r"
expect_prompt

# ==========
# Ensure enumerating functions works if any of them are marked autoloaded but not actually loaded.
# Regression: https://github.com/att/ast/issues/1436
log_test_entry
send "functions\r"
expect -re "\r\ntypeset -fu _cd\r\n.*\r\ntypeset -fu pushd\r\n" {
    puts "typeset -f produces expected output"
}
expect_prompt

# ==========
# Exit the shell.
log_test_entry
send "\r"
expect_prompt
send [ctrl D]
catch {expect default exp_continue} output
log_debug "EOF output: $output"
wait
