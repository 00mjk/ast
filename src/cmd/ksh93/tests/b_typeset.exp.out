typeset -f produces expected output
